Wangscape/utils.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+#include <type_traits>
+#include <stdexcept>
+
+namespace
+{
+
+template<typename I>
+I _pow(I base, I exp)
+{
+    I result = 1;
+    for (int i = 0; i < std::numeric_limits<I>::digits; i++)
+    {
+        if (exp & 1)
+            result *= base;
+        else if (exp == 0)
+            break;
+        exp >>= 1;
+        base *= base;
+    }
+    return result;
+}
+
+}
+
+template <typename I>
+typename std::enable_if<std::is_unsigned<I>::value, I>::type pow(I base, I exp)
```

Comment:
The function should have a comment documenting:
* Template argument requirements.
* Behaviour when invalid arguments are passed (in particular the behaviour for "non-obvious" overflows like `pow<char>(5, 5)`, which is undefined).